thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/performanceEvaluation/IntervalUtils.java

```diff
@@ -0,0 +1,78 @@
+package com.linkedin.thirdeye.anomalydetection.performanceEvaluation;
+
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import org.joda.time.Interval;
+
+
+public class IntervalUtils {
```

Comment:
Joda interval only support overlap, but no function for merge. This logic is similar to the anomaly mergers, but the way using it is quite different. I am not sure if it can be used to TE repository. I will have a further discussion with TE.